### PR TITLE
add a new admin cmd

### PR DESCRIPTION
**Pull request by ethercflow**
Add a new admin command discussed in https://github.com/tikv/tikv/pull/12972#discussion_r1000434376

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### CODE REVIEW

File: proto/pdpb.proto [hunk 1]
```diff
@@ -433,6 +433,10 @@ message SplitRegion {
     repeated bytes keys = 2;
 }
 
+message SwitchWitness {
+    metapb.Peer peer = 1;
```
Comment (BusyJay) [inline proto/pdpb.proto:437]: Seems overwhelming to use `Peer`. How about using ID and flag?

Comment (ethercflow) [reply proto/pdpb.proto]: Sure, using ID and flag instead, thanks!

File: proto/pdpb.proto [hunk 2]
```diff
@@ -433,6 +433,11 @@ message SplitRegion {
     repeated bytes keys = 2;
 }
 
+message SwitchWitness {
+    uint64 id = 1;
```
Comment (Connor1996) [inline proto/pdpb.proto:437]:
```suggestion
    uint64 peer_id = 1;
```

Comment (ethercflow) [reply proto/pdpb.proto]: Fixed, thanks!

**Review by Connor1996 — APPROVED** [review]
LGTM

File: proto/raft_cmdpb.proto [hunk 3]
```diff
@@ -220,7 +220,8 @@ message RollbackMergeRequest {
 message RollbackMergeResponse {}
 
 message SwitchWitnessRequest {
-    metapb.Peer peer = 1;
+    uint64 id = 1;
```
Comment (BusyJay) [inline proto/raft_cmdpb.proto]: Should it be repeated?

File: proto/raft_cmdpb.proto [hunk 4]
```diff
@@ -220,7 +220,8 @@ message RollbackMergeRequest {
...
+    bool isWitness = 2;
```
Comment (BusyJay) [inline proto/raft_cmdpb.proto]: Inconsistent style

File: proto/raft_cmdpb.proto (hunk 3 above)
Comment (ethercflow) [reply proto/raft_cmdpb.proto]: I think should be. eg: distributed across different stores

File: proto/raft_cmdpb.proto (hunk 4 above)
Comment (ethercflow) [reply proto/raft_cmdpb.proto]: consistent, maybe your browser didn't refresh :)

File: proto/raft_cmdpb.proto (hunk 3 above)
Comment (ethercflow) [reply proto/raft_cmdpb.proto]: Fixed, thanks!

File: proto/pdpb.proto [hunk 5]
```diff
@@ -433,6 +433,11 @@ message SplitRegion {
     repeated bytes keys = 2;
 }
 
+message SwitchWitness {
+    repeated uint64 peer_id = 1;
```
Comment (BusyJay) [inline proto/pdpb.proto:437]: Seems extract them as a struct and make the struct repeated is a better style. But this is also OK.

**Review by BusyJay — APPROVED** [review]

File: proto/pdpb.proto (hunk 5 above)
Comment (ethercflow) [reply proto/pdpb.proto:437]: Ok, I plan to change it to `SwitchWitness` and `BatchSwitchWitness` like `Split` and `BatchSplit`.

File: proto/raft_cmdpb.proto [hunk 6]
```diff
@@ -219,6 +219,19 @@ message RollbackMergeRequest {
 
 message RollbackMergeResponse {}
 
+message SwitchWitnessRequest {
+    uint64 peer_id = 1;
+    bool is_witness = 2;
+}
+
+message SwitchWitnessResponse {}
```
Comment (BusyJay) [inline proto/raft_cmdpb.proto:227]: Seems not use at all.

Comment (ethercflow) [reply proto/raft_cmdpb.proto]: Removed, thanks!

**Pull request merged**